uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/staff-leave-report.component.spec.js

```diff
@@ -42,17 +42,19 @@
         spyOn($log, 'debug');
       }));
 
-      beforeEach(inject(function (AbsencePeriodAPIMock, AbsenceTypeAPIMock, EntitlementAPIMock, LeaveRequestAPIMock) {
+      beforeEach(inject(function (AbsencePeriodAPIMock, AbsenceTypeAPIMock, EntitlementAPIMock, LeaveRequestAPIMock, OptionGroupAPIMock) {
         $provide.value('AbsencePeriodAPI', AbsencePeriodAPIMock);
         $provide.value('AbsenceTypeAPI', AbsenceTypeAPIMock);
         $provide.value('EntitlementAPI', EntitlementAPIMock);
         $provide.value('LeaveRequestAPI', LeaveRequestAPIMock);
         $provide.value('checkPermissions', function () { return $q.resolve(isUserAdmin); });
       }));
 
-      beforeEach(inject(['shared-settings', 'HR_settingsMock', function (_sharedSettings_, HRSettingsMock) {
+      beforeEach(inject(['shared-settings', 'HR_settingsMock', 'api.optionGroup.mock', function (_sharedSettings_, HRSettingsMock, _OptionGroupAPIMock_) {
         sharedSettings = _sharedSettings_;
+
         $provide.value('HR_settings', HRSettingsMock);
+        $provide.value('api.optionGroup', _OptionGroupAPIMock_);
```

Comment:
why provide the values for mock here if you are providing values on the previous beforeEach function?

beforeEach - provider
beforeEach - store dependency and mocked services

Reply:
I cannot use _Someting_ notation since the mock name is "api.optionGroup.mock". It is better to keep such in a separate function I guess.